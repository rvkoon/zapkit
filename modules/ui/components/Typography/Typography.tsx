import { createElement, ReactNode } from "react";
import cx from "classnames";
import classNames from "classnames";

export type TypographyComponent =
  | "p"
  | "span"
  | "h1"
  | "h2"
  | "h3"
  | "h4"
  | "h5"
  | "h6";
export type TypographyVariant =
  | "p"
  | "span"
  | "h1"
  | "h2"
  | "h3"
  | "h4"
  | "h5"
  | "h6";
export type TypographyColors = "primary" | "secondary" | "light";

export interface ITypographyProps {
  component?: TypographyComponent;
  variant?: TypographyVariant;
  content: string | ReactNode;
  color?: string;
  className?: string;
}

export function Typography({
  component = "p",
  variant = "p",
  content,
  color = "primary",
  className,
}: ITypographyProps): JSX.Element {
  const isSecondaryColor = color === "secondary";
  const isLightColor = color === "light";

  function renderStyle() {
    switch (variant) {
      case "p":
      case "span":
      default:
        return "text-text-primary";
      case "h1":
        return "text-text-primary text-2xl font-bold";
      case "h2":
        return "text-text-primary text-xl font-bold";
      case "h3":
        return "text-text-primary text-lg font-bold";
      case "h4":
        return "text-text-primary text-m";
      case "h5":
        return "text-text-primary text-m";
      case "h6":
        return "text-text-primary text-m";
    }
  }

  return createElement(component, {
    className: cx(
      renderStyle(),
      isSecondaryColor && "text-text-secondary",
      isLightColor && "text-text-light",
      className
    ),
    children: content,
  });
}
