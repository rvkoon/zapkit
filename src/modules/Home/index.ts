export * from './components';
export * from './helpers';
export * from './hooks';
export * from './services';
export * from './types';
export * from './constants';