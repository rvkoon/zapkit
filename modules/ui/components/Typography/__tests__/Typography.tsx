import { Typography, ITypographyProps } from "../Typography";
import { RenderResult } from "@testing-library/react";
import { render } from "modules/tests";

describe("Typography component", () => {
  const defaultProps: ITypographyProps = {
    content: "typography",
  };

  function renderComponent(props: ITypographyProps): RenderResult {
    return render(<Typography {...props} />);
  }

  it("should render correctly", () => {
    const { container } = renderComponent({ ...defaultProps });
    expect(container).toMatchSnapshot();
  });
});
