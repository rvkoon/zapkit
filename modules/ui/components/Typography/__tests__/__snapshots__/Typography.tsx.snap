// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography component should render correctly 1`] = `
<div>
  <p
    class="text-text-primary"
  >
    typography
  </p>
</div>
`;
